### PR TITLE
`azurerm_analysis_services_server`, `azurerm_mariadb_configuration` : Fix acc tests failing on context no deadline issue

### PR DESCRIPTION
**Pull request by sinbai**
Fix tests [TestAccAzureRMAnalysisServicesServer_suspended ](https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_ANALYSISSERVICES/4980?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildTestsSection=true&expandBuildChangesSection=true) and [TestAccMariaDbConfiguration_](https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_MARIADB/5005?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildTestsSection=true&expandBuildChangesSection=true) failing with error "the context used must have a deadline attached for polling purposes, but got no deadline".

Note: For test `TestAccAzureRMAnalysisServicesServer_suspended ` , the test pass also depends on  [go-azure-sdk PR](https://github.com/hashicorp/go-azure-sdk/pull/606).

### CODE REVIEW

File: internal/services/analysisservices/analysis_services_server_resource_test.go [hunk 1]
```diff
@@ -521,7 +522,10 @@ func (t AnalysisServicesServerResource) suspend(ctx context.Context, clients *cl
 		return err
 	}
 
-	if err := client.SuspendThenPoll(ctx, *id); err != nil {
+	ctx2, cancel := context.WithTimeout(ctx, 15*time.Minute)
```
Comment (stephybun) [inline internal/services/analysisservices/analysis_services_server_resource_test.go:525]: Could we give this a better name 
```suggestion
	timeout, cancel := context.WithTimeout(ctx, 15*time.Minute)
```

File: internal/services/analysisservices/analysis_services_server_resource_test.go [hunk 2]
```diff
@@ -521,7 +522,10 @@ func (t AnalysisServicesServerResource) suspend(ctx context.Context, clients *cl
...
+	defer cancel()
+
+	if err := client.SuspendThenPoll(ctx2, *id); err != nil {
```
Comment (stephybun) [inline internal/services/analysisservices/analysis_services_server_resource_test.go:528]:
```suggestion
	if err := client.SuspendThenPoll(timeout, *id); err != nil {
```

File: internal/services/analysisservices/analysis_services_server_resource_test.go [hunk 3]
```diff
@@ -537,7 +541,10 @@ func (t AnalysisServicesServerResource) checkState(expectedState servers.State)
 			return err
 		}
 
-		resp, err := client.GetDetails(ctx, *id)
+		ctx2, cancel := context.WithTimeout(ctx, 15*time.Minute)
+		defer cancel()
+
+		resp, err := client.GetDetails(ctx2, *id)
```
Comment (stephybun) [inline internal/services/analysisservices/analysis_services_server_resource_test.go:547]:
```suggestion
		timeout, cancel := context.WithTimeout(ctx, 15*time.Minute)
		defer cancel()

		resp, err := client.GetDetails(timeout, *id)
```

File: internal/services/mariadb/mariadb_configuration_resource_test.go [hunk 4]
```diff
@@ -108,7 +109,10 @@ func checkValueIs(value string) acceptance.ClientCheckFunc {
 			return err
 		}
 
-		resp, err := clients.MariaDB.ConfigurationsClient.Get(ctx, *id)
+		ctx2, cancel := context.WithTimeout(ctx, 5*time.Minute)
+		defer cancel()
+
+		resp, err := clients.MariaDB.ConfigurationsClient.Get(ctx2, *id)
```
Comment (stephybun) [inline internal/services/mariadb/mariadb_configuration_resource_test.go:115]:
```suggestion
		timeout, cancel := context.WithTimeout(ctx, 5*time.Minute)
		defer cancel()

		resp, err := clients.MariaDB.ConfigurationsClient.Get(timeout, *id)
```

File: internal/services/mariadb/mariadb_configuration_resource_test.go [hunk 5]
```diff
@@ -138,7 +142,10 @@ func checkValueIsReset(configurationName string) acceptance.ClientCheckFunc {
 
 		id := configurations.NewConfigurationID(serverId.SubscriptionId, serverId.ResourceGroupName, serverId.ServerName, configurationName)
 
-		resp, err := clients.MariaDB.ConfigurationsClient.Get(ctx, id)
+		ctx2, cancel := context.WithTimeout(ctx, 5*time.Minute)
+		defer cancel()
+
+		resp, err := clients.MariaDB.ConfigurationsClient.Get(ctx2, id)
```
Comment (stephybun) [inline internal/services/mariadb/mariadb_configuration_resource_test.go:148]:
```suggestion
		timeout, cancel := context.WithTimeout(ctx, 5*time.Minute)
		defer cancel()

		resp, err := clients.MariaDB.ConfigurationsClient.Get(timeout, id)
```

File: internal/services/analysisservices/analysis_services_server_resource_test.go (hunk 1 above)
Comment (sinbai) [reply internal/services/analysisservices/analysis_services_server_resource_test.go]: fixed.

File: internal/services/analysisservices/analysis_services_server_resource_test.go (hunk 2 above)
Comment (sinbai) [reply internal/services/analysisservices/analysis_services_server_resource_test.go]: fixed.

File: internal/services/analysisservices/analysis_services_server_resource_test.go (hunk 3 above)
Comment (sinbai) [reply internal/services/analysisservices/analysis_services_server_resource_test.go]: fixed.

File: internal/services/mariadb/mariadb_configuration_resource_test.go (hunk 4 above)
Comment (sinbai) [reply internal/services/mariadb/mariadb_configuration_resource_test.go]: fixed.

File: internal/services/mariadb/mariadb_configuration_resource_test.go (hunk 5 above)
Comment (sinbai) [reply internal/services/mariadb/mariadb_configuration_resource_test.go]: fixed.

**Review by stephybun — APPROVED** [review]
Thanks @sinbai LGTM 🐻

**Pull request merged**